### PR TITLE
ci: Add valgrind run

### LINKED ISSUE #17460: Run tests in memory sanitizer
We run the tests in the thread (and other sanitizers), see `git grep sanitizer ./ci`, but not in the memory sanitizer. This seems like it could be improved. See e.g. #17449

### PR DESCRIPTION
**Pull request by maflcko**
Fixes #17460

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK fae4117ddeb5c2264f4d2b1cfcb436b68bd38e86

This has been needed for a very long time! Thanks for adding! :)

**Comment by laanwj** [comment]
ACK fae4117ddeb5c2264f4d2b1cfcb436b68bd38e86

(agree it would be nice to see a failing run as @practicalswift mentions)

File: ci/test/00_setup_env_native_valgrind.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+export LC_ALL=C.UTF-8
+
+export PACKAGES="valgrind clang llvm python3-zmq libssl1.0-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]:
```suggestion
export PACKAGES="valgrind clang llvm python3-zmq libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev"
```

**Comment by elichai** [comment]
I don't see any valgrind output except for the benchmarks https://travis-ci.org/bitcoin/bitcoin/jobs/612624502#L2889

**Comment by MarcoFalke** [comment]
For the unit tests, it will only be printed on failure

**Comment by MarcoFalke** [comment]
Added a failing travis as requested by @practicalswift and @laanwj 

I will remove the failing commit *after* review, but *before* merge.

**Comment by practicalswift** [comment]
ACK facb416ad5ba47b01ee52c273a5c5e3709f06f07 -- diff looks correct

**Comment by MarcoFalke** [comment]
The failing travis build was https://travis-ci.org/bitcoin/bitcoin/builds/623456054

Will reset the branch now

**Comment by MarcoFalke** [comment]
oops, wrong force push

**Comment by MarcoFalke** [comment]
test

**Comment by MarcoFalke** [comment]
test2

**Comment by MarcoFalke** [comment]
:weary: See #17725

**Pull request closed without merging**